index.d.ts

```diff
@@ -860,7 +860,7 @@ export abstract class ReflectionObject {
     public options?: { [k: string]: any };
 
     /** Parsed Options. */
-    public parsedOptions?: { [k: string]: any[] };
+    public parsedOptions?: ({ [k: string]: any | { [innerK: string]: any[] } })[];
```

Comment:
I'm not sure that `any | .....` makes sense.  Also, the second part of the `|` does not look correct to me because according to the test, the value of the nested object does not need to be an array, e.g. `[{"opt1": {a: 1, b: 2}}]`.

Reply:
Hmm, you're right on both counts, not sure why I wrote it as any[] 🤔 

I think that the best option here is then to get rid of the `|` part, and leave it as `({ [k: string]: any})[]`
Trying to describe every option with typescript would probably be more confusing than helpful

Reply:
I would just leave it as `any`, yes. If there's a type for "any proto message" (like a nested object with values that can be arrays, objects, strings, numbers, booleans, nulls, etc. :) ) defined somewhere, we can use it here, but it won't be too helpful I guess.